### PR TITLE
Introduce SSAI API

### PR DESCRIPTION
**Pull request by strangesource**
## Problem
SSAI ad tracking is not yet supported

## Changes
- **Introduce new SSAI API without using it**
- **Add unit tests for SSAI API**

The following files are public API:
- [conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/AdPosition.java](https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/pull/72/files#diff-2b72a24c15b769f4ad13aaf8c38dfd17b334bc5ab062f3bc59bf9329e028a8ac)
- [conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/SsaiApi.java](https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/pull/72/files#diff-09f0b789cdf2f6d9ba79f472acc710306be834ef2d20d0c290153cef995ab8a6)

## Related Changes
- #69 
- #70

### CODE REVIEW

**Review by rolandkakonyi — APPROVED** [review]
Very nice implementation!

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/SsaiApi.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+     * Reports the end of a server-side ad break. Must be called after the last ad has finished.
+     * Has no effect if no server-side ad break is currently active.
+     */
+    void reportAdBreakFinished();
+
+    /**
+     * Reports the start of a server-side ad.
+     *
+     * @param adInfo Object containing metadata about the server-side ad.
+     */
```
Comment (matamegger) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/SsaiApi.java:40]: Has to be called after `reportAdBreakStarted`, right?

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/SsaiApi.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
+package com.bitmovin.analytics.conviva.ssai;
+
+import java.util.Map;
+
+/**
+ * Enables reporting of server-side ad breaks and ads.
+ */
+public interface SsaiApi {
+    /**
+     * Checks if a server-side ad break is currently active.
```
Comment (matamegger) [inline conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/SsaiApi.java:10]:
```suggestion
     * Reports if a server-side ad break is currently active.
```

**Review by matamegger — APPROVED** [review]
Very nice! Just some questions and minor remarks.

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/SsaiApi.java (hunk 1 above)
Comment (strangesource) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/SsaiApi.java:40]: Yes, I added it to the documentation.

Comment (strangesource) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/SsaiApi.java:40]: [9a52f65](https://github.com/bitmovin/bitmovin-player-android-analytics-conviva/pull/72/commits/9a52f6526daf245ee6f81cb8200664934a101409)

File: conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/DefaultSsaiApi.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    @Override
+    public void reportAdBreakStarted() {
+        reportAdBreakStarted(null);
+    }
+
+    public void reset() {
+        isAdBreakActive = false;
+    }
```
[the remark this answers is not in the document]
Comment (strangesource) [reply conviva/src/main/java/com/bitmovin/analytics/conviva/ssai/DefaultSsaiApi.java:39]: We don't hold any state on our side and whatever is in the ad analytics will be reset in the `ConvivaAnalyticsIntegration`.

**Pull request merged**